library/common/engine.h

```diff
@@ -63,6 +63,13 @@ class Engine {
    */
   envoy_status_t recordGaugeSub(const std::string& elements, uint64_t amount);
 
+  /**
+   * Record a value for the histogram with the given string of elements
+   * @param elements, joined elements of the timeseries.
+   * @param amount, amount to add to the aggregated pool of events for quantile calculations
+   */
+  envoy_status_t recordHistogramDurationMs(const std::string& elements, uint64_t amount);
```

Comment:
Put this down as a DurationMs since it's the most common use case I see on the mobile side, but I'm going to move this to an enum so clients can record any unit values e.g. bytes/ms/us

Reply:
probably just do `recordHistogram` here, and client can decide the unit for the histogram

Reply:
and `uint64_t amount` -> `uint64_t value`